ray-curator/ray_curator/stages/text/modules/score_filter.py

```diff
@@ -182,70 +255,120 @@ def process(self, batch: DocumentBatch) -> DocumentBatch | None:
+        invert (bool | list[bool]): If True, will keep all documents that are normally discarded.
 
     """
 
-    filter_obj: DocumentFilter
-    text_field: str = "text"
-    score_field: str | None = None
-    invert: bool = False
+    filter_obj: DocumentFilter | list[DocumentFilter]
+    text_field: str | list[str] = "text"
+    score_field: str | list[str] | None = None
+    invert: bool | list[bool] = False
     _name: str = "score_filter"
 
     def __post_init__(self):
-        self._name = self.filter_obj.name
+        if isinstance(self.filter_obj, list):
+            self._name = "score_filter_chain"
+
+            num_filters = len(self.filter_obj)
+
+            # Okay to assume same text field and invert flag for all filters
+            if isinstance(self.text_field, str):
+                logger.info(f"Using the same text field for all filters: {self.text_field}")
+                self.text_field = [self.text_field] * num_filters
+            if self.invert is not None and isinstance(self.invert, bool):
+                logger.info(f"Using the same invert flag for all filters: {self.invert}")
+                self.invert = [self.invert] * num_filters
+
+            # Score field must be a list of strings if multiple filters are used
+            # Otherwise, the field will be overwritten for each filter
+            if self.score_field is not None and isinstance(self.score_field, str):
+                msg = f"Score field must be a list of strings if multiple filters are used: {self.score_field}"
+                raise ValueError(msg)
+            elif self.score_field is None:
+                self.score_field = [None] * num_filters
+
+        else:
+            self._name = self.filter_obj.name
+
+            if isinstance(self.text_field, list) and len(self.text_field) > 1:
+                msg = f"More text fields than filters provided: {self.text_field}"
+                raise ValueError(msg)
+            if self.score_field is not None and isinstance(self.score_field, list) and len(self.score_field) > 1:
+                msg = f"More score fields than filters provided: {self.score_field}"
+                raise ValueError(msg)
+            if isinstance(self.invert, list) and len(self.invert) > 1:
+                msg = f"More invert flags than filters provided: {self.invert}"
+                raise ValueError(msg)
+
+            self.filter_obj = [self.filter_obj]
+            self.text_field = [self.text_field]
+            self.score_field = [self.score_field]
+            self.invert = [self.invert]
 
     def inputs(self) -> tuple[list[str], list[str]]:
-        return ["data"], [self.text_field]
+        return ["data"], self.text_field
 
     def outputs(self) -> tuple[list[str], list[str]]:
-        return ["data"], [self.score_field] if self.score_field else []
+        return ["data"], self.score_field if self.score_field is not None else []
```

Comment:
Should we add validation that text field is not dropped here too ? 
```suggestion
        return ["data"], [self.text_field] + (self.score_field if self.score_field is not None else [])
```